packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1690,16 +1689,567 @@ async function prerenderToStream(
+
+            // We don't actually want to render anything so we just pass a stream
+            // that never resolves. The resume call is going to abort immediately anyway
+            const foreverStream = new ReadableStream<Uint8Array>()
+
+            const resumeStream = await resume(
+              <App
+                reactServerStream={foreverStream}
+                preinitScripts={() => {}}
+                clientReferenceManifest={clientReferenceManifest}
+                ServerInsertedHTMLProvider={ServerInsertedHTMLProvider}
+                nonce={ctx.nonce}
+              />,
+              JSON.parse(JSON.stringify(postponed)),
+              {
+                signal: createPostponedAbortSignal('static prerender resume'),
+                onError: htmlRendererErrorHandler,
+                nonce: ctx.nonce,
+              }
+            )
+
+            // First we write everything from the prerender, then we write everything from the aborted resume render
+            htmlStream = chainStreams(prelude, resumeStream)
+          }
+
+          return {
+            digestErrorsMap: reactServerErrorsByDigest,
+            ssrErrors: allCapturedErrors,
+            stream: await continueStaticPrerender(htmlStream, {
+              inlinedDataStream: createInlinedDataReadableStream(
+                reactServerResult.getStream(),
+                ctx.nonce,
+                formState
+              ),
+              getServerInsertedHTML,
+            }),
+            dynamicTracking,
+          }
+        }
+      } else {
+        /**
+         * dynamicIO without PPR
+         *
+         * The general approach is to render the RSC tree first allowing for any inflight
+         * caches to resolve. Once we have settled inflight caches we can check and see if any
+         * synchronous dynamic APIs were used. If so we don't need to bother doing anything more
+         * because the page will be dynamic on re-render anyway
+         *
+         * If no sync dynamic APIs were used we then re-render and abort after a single Task.
+         * If the render errors we know that the page has some dynamic IO. This assumes and relies
+         * upon caches reading from a in process memory cache and resolving in a microtask. While this
+         * is true from our own default cache implementation and if you don't exceed our LRU size it
+         * might not be true for custom cache implementations.
+         *
+         * Future implementations can do some different strategies during build like using IPC to
+         * synchronously fill caches during this special rendering mode. For now this heuristic should work
+         */
+
+        const cache = staticGenerationStore.incrementalCache
+        if (!cache) {
+          throw new Error(
+            'Expected incremental cache to exist. This is a bug in Next.js'
+          )
+        }
+
+        const PRERENDER_COMPLETE = 'NEXT_PRERENDER_COMPLETE'
```

Comment:
this should probably use the same definition on line 1708 rather than re-declaring it